junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/AbstractExtensionContext.java

```diff
@@ -53,9 +56,10 @@ abstract class AbstractExtensionContext<T extends TestDescriptor> implements Ext
 	private final JupiterConfiguration configuration;
 	private final NamespacedHierarchicalStore<Namespace> valuesStore;
 	private final ExecutableInvoker executableInvoker;
+	private final ExtensionRegistry extensionRegistry;
 
 	AbstractExtensionContext(ExtensionContext parent, EngineExecutionListener engineExecutionListener, T testDescriptor,
-			JupiterConfiguration configuration,
+			JupiterConfiguration configuration, ExtensionRegistry extensionRegistry,
 			Function<ExtensionContext, ExecutableInvoker> executableInvokerFactory) {
 		this.executableInvoker = executableInvokerFactory.apply(this);
```

Comment:
Now that we pass in the `ExtensionRegistry`, we should be able to remove the `Function` parameter and instantiate `DefaultExecutableInvoker` directly.

```suggestion
			JupiterConfiguration configuration, ExtensionRegistry extensionRegistry) {

		this.executableInvoker = new DefaultExecutableInvoker(this, extensionRegistry);
```